web/lib/websocket/game-loop.js

```diff
@@ -0,0 +1,357 @@
+module.exports = GameLoop;
+var
+  Promise = require('bluebird');
+  models = require('../../models');
+
+function GameLoop(socket, msg) {
+  this.socket = socket;
+  this.msg = msg;
+  this.round = null;
+  this.judge = null;
+  this.game = null;
+}
+
+GameLoop.prototype.getGameId = function() {
+  var self = this;
+  return new Promise(function(resolve, reject) {
+    models.Game.find({
+      where: {
+        roomId: room,
+        finishTime: null
+      } 
+    }).then(function(game) {
+      self.game = game.id; 
+      resolve();
+    }); 
+  });
+}
+
+GameLoop.prototype.getCurrentRoundAndJudge = function() {
+  var self = this;
+  return new Promise(function(resolve, reject) {
+    models.Round.find({
+      where: {
+        gameId: game,
+        state: {
+          $not: "over" 
+        }
+      }
+    }).then(function(round) {
+      self.round = round.id; 
+      self.judge = round.judge;
+    }); 
+  });
+}
+
+/*
+  -Create PlayedCard
+  -update Hand entry to played=true
+  -update PlayerState to "waiting for players"
+  -check if all players are waiting
+    
+  If all players are waiting
+  -move to play time up
+  else
+  -just broadcast that this player played a card
+
+  msg is of the form:
+  {
+     card: id of card played
+  }
+ */
+GameLoop.prototype.handlePlay = function() {
+  var self = this;
+
+  var socket = self.socket;
+  var user = socket.userId;
+  var room = socket.roomId;
+
+  var msg = self.msg;
+  var card = msg.card;
+
+  self.getGameId()
+  .then(self.getCurrentRoundAndJudge)
+  .then(createPlayedCard)
+  .then(updateHandToPlayed)
+  .then(updatePlayerStateToWaiting)
+  .then(updateGameState)
+  .then(broadcastResponse)
+  .catch(function(errors) {
+    socket.emit('error', errors);
+    reject(errors)
+  });
+
+  
+  function createPlayedCard() {
+    return new Promise(function(resolve, reject) {
+      models.PlayedCard.create({
+        userId: user,
+        cardId: card,
+        roundId: round
+      }).then(function(playedCard) {
+        resolve();
+      })
+    });
+  }
+
+  function updateHandToPlayed() {
+    return new Promise(function(resolve, reject) {
+      var values = {
+        played: true 
+      };
+
+      models.Hand.find({
+        where: {
+          cardId: card,
+          gameId: game,
+          userId: user,
+          played: false
+        } 
+      }).then(function(hand) {
+        models.Hand.update(values, {
+          where: {id: hand}
+        }).then(function(count, obj) {
```

Comment:
This update call looks fucked up.

I think you can just delete the where parameter.